x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2855 @@
+
+During this RFC, we will explicitly refer to case of Elastic prebuilt rules, but take into account that this will be only a specific case of all the possible external sources for rules.
+
+## Necessary rule schema changes
+
+In order to support the customization of Elastic Prebuilt Rules, we need to modify our rule schema. This involves introducing the new top level field: the nested `rule_source` field, as well deprecating the `immutable` field.
+
+```ts
+// PSEUDOCODE - see complete schema in detail below
+{
+  rule_source: {
+    type: 'external'
+    is_customized: boolean;
+    source_updated_at?: Date;
+  } | {
+    type: 'internal'
+  },
+}
+```
+
+
+### `rule_source` field
+
+The `rule_source` field will be a required top-level object field in our rule schema that will be a discriminated union of two types: `'internal'` and `'external'`.
+
+Rules with `rule_source` of type `internal` are rules generated internally in the Kibana application and which don't have an external origin outside of it.
+
+Rules with `rule_source` of type `external` are rules who have an external origin or source. Elastic prebuilt rules are a case of `external` rule, with the `detection-rules` repository handled by the TRaDE team being their external source origin.
+
+This also means that a rule with this type of `rule_source` will determine that the rule is an Elastic Prebuilt Rule that can receive upstream updates via Fleet. This field is intented to partly replace the currently existing `immutable` field, which is used today for the same purpose, but -as its name indicates- also currently determines if a rule's fields can be modified/customized.
+
+As seen in the schema, when the `rule_source` has type `'external'`, the object can contain two other subfields as well, `is_customized` and `source_updated_at`, explained below. If the object has type `'internal'`, no other fields should be present (subject to change in the future).
+
+### Deprecating the `immutable` field
+
+In the current application's state, rules with `immutable: false` are rules which are not Elastic Prebuilt Rules, i.e. custom rules, and can be modified. Meanwhile, `immutable: true` rules are Elastic Prebuilt Rules, created by the TRaDE team, distributed via the `security_detection_engine` Fleet package, and cannot be modified once installed.
+
+When successfully implemented, the `rule_source` field should replace the `immutable` field as a mechanism to mark Elastic prebuilt rules, but with one difference: the `rule_source` field will determine if the rule is an Elastic Prebuilt Rule or not, but now all rules will be customizable by the user in Kibana, i.e. independently of the existence (or absence) of `rule_source`.
```

Comment:
Would it be more generic to use term `external rules` instead of Elastic Prebuilt Rules? Since we want to take into account Detection-as-Code general terms will help to keep the document relevant and avoid confusion as long as possible.

Reply:
There's a note at the begginning of the RFC that explains this: we want to keep the discussion centered around prebuilt rules, although the changes that we are making are as forward-compatible as possible with DoC.

Reply:
@jpdjere sorry, I didn't get if your answer is yes or no. If it's yes to simplify the document Elastic Prebuilt Rules can be omitted everywhere besides the document beginning and replaced with external rules.

Reply:
The answer is no. We want to refer to Elastic prebuilt rules for now as this is what the epic relates to. When we start dealing with DaC, we can generalize the term to "external rules".